apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -1451,7 +1451,9 @@ impl FetchDependencyGraph {
         // subgraph name, but have no worries for `mergeAt` since it contains either number of
         // field names, and the later is restricted by graphQL so as to not be an issue.
         let mut by_subgraphs = MultiMap::new();
-        for node_index in self.graph.node_indices() {
+        let sorted_nodes = petgraph::algo::toposort(&self.graph, None)
+            .map_err(|_| FederationError::internal("Failed to sort nodes due to cycle(s)"))?;
+        for node_index in sorted_nodes {
```

Comment:
non-blocking comment: what's the performance cost of doing a topological sort here, do you know?

Reply:
I tried a few queries with known-big fetch dependency graphs, and this function doesn't really show up in a flame graph. I think because it only happens once, as the absolute last step, when the graph has already been reduced as much as it can.

Reply:
excellent excellent!!

Reply:
Thanks for that analysis. I didn't observe any noticeable slowdown during corpus comparison runs, either.